### ISSUE TITLE
HDK3 cleanup tasks

### ISSUE THREAD

**Issue by pospi** [holo-rea/holo-rea#174]
The migration to "Holochain RSM" is now complete-enough to be considered a new stable foundation to integrate from. A prototype version compatible with the previous HDK2 is tagged at the [hdk-2.x branch](https://github.com/holo-rea/holo-rea/tree/hdk-2.x).

Filing this issue to track cleanup jobs needed to finalise the task and achieve feature parity with the old version-

### Blocking

- [x] #213
- [x] #218
- [x] get GraphQL explorer wired up again (pending resolution of above)
    - [x] update conductor running command

### Bugs

- [x] fix all `EconomicResource` container refs being cleared when only 1 is removed
- [x] ensure `EconomicResource` stage & state still work (they query the DHT for related records directly...)
- [x] ensure unit CRUD (and "anchored records") work as expected after changes to indexing architecture

### Code clarity & maintainability / repo organisation / cleanup

- [x] #216
- [x] #267
- [ ] #214
- [ ] #215
- [x] move all indexing library code to its own crates, and split out index helpers into own `hdk_semantic_indexes` crate that `hdk_records` can depend upon
- [x] change all `generate_query_handler` methods to a more abstract macro that can be statically compiled
- [x] simplify / consolidate naming of indexing helper methods, since all index types now have a coherent API
- [x] cleanup some residual zome API code duplication & misplaced structs:
    - WASM API parameter structs should be declared in `rpc/`
    - `ByHeader` & `ByAddress` only declared once; `ByHeader.revision_id` instead of `.address`

### Administration & documentation

- publish Nodejs modules
    - [x] `@valueflows/vf-graphql-holochain`
    - [x] `@vf-ui/graphql-client-holochain`
- [x] #70 
- update workflow automation docs
- update [DNA Auth Resolver](https://github.com/holochain-open-dev/dna-auth-resolver/) readme to point to final location of `hdk_records` lib, publish `hc_zome_dna_auth_resolver` crates

### Work ahead and other things to note

- This upgrade brings us a long way towards #49 as all record identifiers now include a `DnaHash`
- There's stuff to leverage in terms of the Agent module, eg. #172
- Revisit ID field encoding someday #173

**Comment by artbrock** [comment]
## RE: Blocking Issues

### hREA is too large to install (core issue)
 - Unable to install large apps holochain/holochain#966
 
I responded to this issue. I'm pretty sure there is an existing workaround in that mr_bundle's location format for any bundled resource (zomes, dnas, etc.) includes options for file_paths and web URLs/URIs so you should be able to make bundles smaller than 16MB that can then pull their components from a specified location.

- get GraphQL explorer wired up again (pending resolution of above)
    - update conductor running command

I'm not sure what you're asking of us for these items. Can you clarify?

**Comment by artbrock** [comment]
Hehe... Oops... I followed a link to get here and thought you were reporting HDK3 issues to Holochain, not tracking your own update issues... Once I made it further down the list, I got it. 

I hope using other formats for location references unblocks you!

**Comment by Connoropolous** [comment]
anything in this epic issue that's not already done has an open issue for itself, and it feels like we've cleared this one and everything it captured! Closing
